stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/SocketFactory.kt

```diff
@@ -71,9 +71,15 @@ internal class SocketFactory(
         private const val ANONYMOUS_USER_ID = "anon"
     }
 
-    private fun User.reduceUserDetails() = this.copy(
-        devices = emptyList(),
-        mutes = emptyList(),
-        channelMutes = emptyList()
-    )
+    private fun User.reduceUserDetails(): Map<String, Any> {
+        val details = mutableMapOf(
+            "id" to id,
+            "role" to role,
+            "banned" to banned,
+            "invisible" to invisible,
+            "teams" to teams,
+        )
+        details.putAll(extraData)
+        return details
+    }
```

Comment:
This change makes this a pure map that's passed to the serialization code, instead of containing a domain model `User` object (which only the Gson impl could deal with)